beacon-chain/blockchain/forkchoice_update_execution.go

```diff
@@ -0,0 +1,61 @@
+package blockchain
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v3/beacon-chain/state"
+	"github.com/prysmaticlabs/prysm/v3/consensus-types/interfaces"
+)
+
+func (s *Service) isNewProposer() bool {
+	_, _, ok := s.cfg.ProposerSlotIndexCache.GetProposerPayloadIDs(s.CurrentSlot()+1, [32]byte{} /* root */)
+	return ok
+}
+
+func (s *Service) isNewHead(r [32]byte) bool {
```

Comment:
This adds duplicated logic in a few places. At the very least I remember `saveHead` where the same check its used. If we are going to have a dedicated method to check if a given block root differs from the stored head, we should move these checks to use this method. 

Also, the logic in `saveHead` is a little safer with respect to the origin Root, so I'll extend slightly this comment. This is what `saveHead` currently has:
```go
func (s *Service) saveHead(ctx context.Context, newHeadRoot [32]byte, headBlock interfaces.SignedBeaconBlock, headState state.BeaconState) error {
...
	var oldHeadRoot [32]byte
	s.headLock.RLock()
	if s.head == nil {
		oldHeadRoot = s.originBlockRoot
	} else {
		oldHeadRoot = s.head.root
	}
	s.headLock.RUnlock()
	if newHeadRoot == oldHeadRoot {
		return nil
	}
```
We should move the last check to 
```go
if s.isNewHead(newHeadRoot)
```
But also, we should probably add the checks for head being nil and dealing with the `originBlockRoot` in this method as well.

Reply:
Done. One annoying part is we still need to retrieve `oldHeadRoot` for logs